src/Symfony/Component/Form/Extension/Core/Type/FormType.php

```diff
@@ -33,10 +33,13 @@ class FormType extends BaseType
 
     public function __construct(?PropertyAccessorInterface $propertyAccessor = null)
     {
-        $this->dataMapper = new DataMapper(new ChainAccessor([
+        $propertyPathAccessor = new PropertyPathAccessor($propertyAccessor ?? PropertyAccess::createPropertyAccessor());
+        $dataAccessor = new ChainAccessor([
             new CallbackAccessor(),
-            new PropertyPathAccessor($propertyAccessor ?? PropertyAccess::createPropertyAccessor()),
-        ]));
+            $propertyPathAccessor,
+        ]);
+        $propertyPathAccessor->setReadAccessor($dataAccessor);
```

Comment:
Too bad this creates a circular ref. Any way around?

(don't miss adding a test case of course ;) )

Reply:
I thought about it too, but it's only possible if someone configures the chain on their own; otherwise, it will call the getter accessor.

Reply:
With the current architecture there is no way around it. If we were to design the interface again with the knowledge we have now, we would probably do something like optionally passing a read accessor around when calling `setValue()`.

Reply:
I think I found a better solution that does not involve creating the circular reference